docs/guides/go.md

```diff
@@ -311,112 +310,89 @@ information on why the transaction was rejected.
-receive a block.
-- `DeliverTx` is called repeatedly, once for each application transaction that was included in the block.
-- `EndBlock` is called once to indicate to the application that no more transactions
-will be delivered to the application in within this block.
+`FinalizeBlock` is a method introduced in CometBFT v0.38.0 which replaces the traditional ABCI methods of `BeginBlock`, `DeliverTx`, and `EndBlock`.
 
-Note that, to implement these calls in our application we're going to make use of Badger's
-transaction mechanism. We will always refer to these as Badger transactions, not to
-confuse them with the transactions included in the blocks delivered by CometBFT,
-the _application transactions_.
+This method is responsible for executing the block and returning a response to the consensus engine. By implementing `ProcessProposal`, transactions are processed before the block execution phase and are already present in the state machine. This means that there is no need for the traditional ABCI methods of `BeginBlock`, `DeliverTx`, and `EndBlock` to handle the transaction execution and state updates. This simplifies the ABCI interface and increases flexibility in the execution pipeline.
 
-First, let's create a new Badger transaction during `BeginBlock`. All application transactions in the
-current block will be executed within this Badger transaction.
-Then, return informing CometBFT that the application is ready to receive application transactions:
+The `FinalizeBlock` method executes the block, including any necessary transaction processing and state updates, and returns a `ResponseFinalizeBlock` object which contains any necessary information about the executed block.
 
 ```go
-func (app *KVStoreApplication) BeginBlock(req abcitypes.RequestBeginBlock) abcitypes.ResponseBeginBlock {
-	app.onGoingBlock = app.db.NewTransaction(true)
-	return abcitypes.ResponseBeginBlock{}
+func (app *KVStoreApplication) FinalizeBlock(_ context.Context, req *abcitypes.RequestFinalizeBlock) (*abcitypes.ResponseFinalizeBlock, error) {
+    var txs = make([]*abcitypes.ExecTxResult, len(req.Txs))
+
+    app.onGoingBlock = app.db.NewTransaction(true)
+    for i, tx := range req.Txs {
+        if code := app.isValid(tx); code != 0 {
+            log.Printf("Error in tx in if")
+            txs[i] = &abcitypes.ExecTxResult{Code: code}
+        } else {
+            parts := bytes.SplitN(tx, []byte("="), 2)
+            key, value := parts[0], parts[1]
+            log.Printf("Adding key %s with value %s", key, value)
+
+            if err := app.onGoingBlock.Set(key, value); err != nil {
+                log.Panicf("Error writing to database, unable to execute tx: %v", err)
+            }
+            log.Printf("Successfully added key %s with value %s", key, value)
+
+            // Marshal the ExecTxResult to the txs slice element
+            txs[i] = &abcitypes.ExecTxResult{}
+            if _, err := txs[i].MarshalToSizedBuffer(nil); err != nil {
```

Comment:
Why do we need to do this check here? If we need to serialize and it fails, we'll panic at that point